src/main/java/sg/ncl/MainController.java

```diff
@@ -5150,6 +5189,34 @@ private int getNodes(NodeType nodeType) {
         return usageInfoList;
     }
 
+    private Long getTeamUsageStatistics(Team2 team2, String start, String end, HttpEntity<String> request, List<Long> usages) throws Exception {
+        Long usage = 0L;
+        ResponseEntity responseEntity = restTemplate.exchange(properties.getUsageStat(team2.getId(), "startDate=" + start, "endDate=" + end), HttpMethod.GET, request, String.class);
+        String responseBody = responseEntity.getBody().toString();
+        JSONArray jsonArray = new JSONArray(responseBody);
+
+        // handling exceptions from SIO
+        if (RestUtil.isError(responseEntity.getStatusCode())) {
+            MyErrorResource error = objectMapper.readValue(responseBody, MyErrorResource.class);
+            ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
+            switch (exceptionState) {
+                case START_DATE_AFTER_END_DATE_EXCEPTION:
+                    log.warn("Get team usage : Start date after end date error");
+                    throw new StartDateAfterEndDateException(ERR_START_DATE_AFTER_END_DATE);
+                default:
+                    log.warn("Get team usage : sio or deterlab adapter connection error");
+                    throw new Exception(ERR_SERVER_OVERLOAD);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)